src/nats.rs

```diff
@@ -161,3 +133,148 @@ pub(crate) fn from_tls_auth_config(
         }
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    fn parse_auth(s: &str) -> Result<nats::asynk::Options, crate::Error> {
+        toml::from_str(s)
+            .map_err(Into::into)
+            .and_then(|config: NatsAuthConfig| config.to_nats_options().map_err(Into::into))
+    }
+
+    #[test]
+    fn auth_user_password_ok() {
+        parse_auth(
+            r#"
+            strategy = "user_password"
```

Comment:
Thoughts on changing this to `basic` to match other components (ex. `http`)?

Reply:
I would tend to disagree. AFAICT NATS isn't using HTTP authentication, and their documentation does not use the "basic" terminology to describe this mode.

Reply:
That's something I thought myself as well.